superset-frontend/src/common/components/Tabs.tsx

```diff
@@ -16,27 +16,53 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import React from 'react';
 import { styled } from '@superset-ui/core';
 import { Tabs as AntdTabs } from 'src/common/components';
+import { css } from '@emotion/core';
+import Icon from '../../components/Icon';
 
-const StyledTabs = styled(AntdTabs)`
-  margin-top: -18px;
+interface TabsProps {
+  inModal?: boolean;
+  fullWidth?: boolean;
+}
 
-  .ant-tabs-nav-list {
-    width: 100%;
-  }
+const notForwardedProps = ['fullWidth', 'inModal'];
+
+const StyledTabs = styled(AntdTabs, {
+  shouldForwardProp: prop => !notForwardedProps.includes(prop),
+})<TabsProps>`
+  ${props =>
+    props.inModal &&
```

Comment:
These styles should probably be added to the Modal, since it's more the concern of the modal, and the Tabs shouldn't be concerned where they live.

Reply:
Done